Update the gprofiler module to g:P2 API

tambeta opened the pull request:
Make requests against the new JSON API. Warn about deprecated attributes.

tambeta committed:
Replace the deprecated gulp-utils package

tambeta committed:
Update GProfiler attribute parsing to gP:2 API

Make POST requests with a JSON payload against the new API endpoint.
Warn about deprecated attributes.

tambeta committed:
Update GProfiler result parsing to g:P2 API

tambeta committed:
GProfiler: differentiate API and app root URLs

tambeta committed:
Update Travis CI config

tambeta committed:
DOM tests: render from stored payload

Make DOM tests better contained by removing dependencies on the remote
server and the ability of PhantomJS to handle the SSL protocol. This
actually tests what happens in the DOM instead of integration testing.

Add the package lock and mark some uninteresting files as binary in
.gitattributes.

tambeta committed:
Remove package-lock.json as it trips up Travis

See e.g.: https://github.com/gerbera/gerbera/issues/253

tambeta committed:
gulpfile: use single quotes throughout

tambeta committed:
Fix the docs builder + README.md

Header markers in jsdoc output are now separated by a space from the
header text for valid Markdown.

tambeta committed:
README: fix a couple internal links

tambeta committed:
Bump version to 0.6.0, update docs

tambeta committed:
Remove option to make a POST request on logo click

Update g:Profiler logo, make URL construction return early if length
limit exceeded.

tambeta committed:
Minor README fix #2

tambeta committed:
Minor README fix

urdvr commented:
I built and tested against a local version of g:Profiler. 

As far as I can tell, `BioJSVisGProfiler` could make use of calling g:Profiler with `"no_evidences":true` as it doesn't make use of evidence codes, but it could be a case of unneeded optimization

Looks fine to merge by me.

tambeta committed:
Query g:GOSt with no_evidences = true

GProfiler currently does not make use of intersection / evindence data.
Optimize the query by hardcoding `no_evidences` to true.

tambeta committed:
Fix linter error for previous commit

tambeta commented:
Merged, closed and published.
